backend/layers/persistence/persistence_mock.py

```diff
@@ -159,7 +165,9 @@ def add_collection_version(self, collection_id: CollectionId) -> CollectionVersi
         return new_version_id
 
     def delete_collection_version(self, version_id: CollectionVersionId) -> None:
-        # Can only delete an unpublished collection
+        collection_version = self.collections_versions.get(version_id.id)
+        if collection_version.published_at:
```

Comment:
I generally prefer an explicit `is not None`. Even if this is a super edge case, a collection published in January 1st, 1970 at midnight would result as non published 😄